cwms_radar_api/src/main/java/cwms/radar/data/dao/LocationsDaoImpl.java

```diff
@@ -248,186 +250,129 @@ public static Feature buildFeatureFromAvLocRecord(Record avLocRecord) {
-                    .innerJoin(AV_LOC_GRP_ASSGN).on(
-                            AV_LOC.LOCATION_ID.eq(AV_LOC_GRP_ASSGN.LOCATION_ID))
-                    .where(condition);
+                .from(AV_LOC2.AV_LOC2)
+                .where(condition);
 
             total = count.fetchOne().value1();
         } else {
             // get totally from page
-            String[] parts = CwmsDTOPaginated.decodeCursor(cursor, "|||");
-
-            if (parts.length > 1) {
-                locCursor = parts[0].split("/")[1];
-                total = Integer.parseInt(parts[1]);
-            }
+            Catalog.CatalogPage page = new Catalog.CatalogPage(cursor);
+            locCursor = page.getCursorId();
+            total = page.getTotal();
         }
 
-        Condition condition = buildCatalogWhere(unitSystem, office, idLike, categoryLike, groupLike)
-                .and(AV_LOC.LOCATION_ID.upper().greaterThan(locCursor));
-
-        SelectConditionStep<Record1<String>> tmp = dsl.selectDistinct(AV_LOC.LOCATION_ID)
-                .from(AV_LOC).innerJoin(AV_LOC_GRP_ASSGN).on(
-                        AV_LOC.LOCATION_ID.eq(AV_LOC_GRP_ASSGN.LOCATION_ID))
-                .where(condition);
-
-        Table<?> forLimit = tmp.orderBy(AV_LOC.LOCATION_ID).limit(pageSize).asTable();
-
-        SelectConditionStep<Record> query = dsl.select(
-                        AV_LOC.asterisk(),
-                        AV_LOC_GRP_ASSGN.asterisk()
-                )
-                .from(AV_LOC)
-                .innerJoin(forLimit).on(forLimit.field(AV_LOC.LOCATION_ID).eq(AV_LOC.LOCATION_ID))
-                .leftJoin(AV_LOC_GRP_ASSGN).on(AV_LOC_GRP_ASSGN.LOCATION_ID.eq(AV_LOC.LOCATION_ID))
-                .where(condition);
-
-        query.orderBy(AV_LOC.LOCATION_ID);
-//        logger.info( () -> query.getSQL(ParamType.INLINED));
-
-        Map<usace.cwms.db.jooq.codegen.tables.records.AV_LOC,
-                Set<LocationAlias>> theMap = new LinkedHashMap<>();
-
-
-        query.fetch().forEach(row -> {
-            usace.cwms.db.jooq.codegen.tables.records.AV_LOC loc = buildAvLoc(row);
-
-            Set<LocationAlias> locationAliases = theMap.computeIfAbsent(loc,
-                    k -> new LinkedHashSet<>());
-
-            locationAliases.add(buildLocationAlias(row));
-        });
-
-        List<? extends CatalogEntry> entries =
-                theMap.entrySet().stream()
-                        .sorted((left, right) -> (
-                                left.getKey().getLOCATION_ID()
-                                        .compareTo(right.getKey().getBASE_LOCATION_ID())
-                        ))
-                        .map(e -> buildCatalogEntry(e.getKey(), e.getValue())
-                        ).collect(Collectors.toList());
-
+        condition = condition.and(AV_LOC2.AV_LOC2.LOCATION_ID.upper().greaterThan(locCursor));
+
+        SelectLimitPercentStep<Record1<Long>> forLimit = dsl.select(AV_LOC2.AV_LOC2.LOCATION_CODE)
```

Comment:
This should get a fixed name. I've noticed that JOOQ will often build a new query each time and give a random identifier. Reeks havoc on the shared memory for plans.

Reply:
True of any subselect.

Reply:
Thanks for the catch, your warning from awhile back is now ringing in my ears.